### ISSUE TITLE
Fix: Storybook Accessibility issues

### ISSUE THREAD

**Issue by nikitaorliak-cengage** [cengage/react-magma#1973]
**Describe the bug**
Fix all storybook Accessibility issues 

### Issue 6 -- https://github.com/cengage/react-magma/issues/1859#issuecomment-3385935835

**Acceptance Criteria**
Go through each of the issues listed above and determine:
1. Do we want to fix it? If "easy", fix immediately. If "hard" (more than a day's worth of work), separate into new ticket
2. Is it valid? If not, cross out from the list
3. For color contrast issues, work with Orion

**Comment by artemtrusov-cengage** [comment]
- ❌ — needs a fix
- ✅ — fixed
- ❓ — issue still present, not sure we want to fix it
- 🚀 — quick fix
- ⏰ — needs a new ticket



---
**Part 1**
---


- **Issue 1** -- https://github.com/cengage/react-magma/issues/1856#issuecomment-3370411781  ✅

    Axe in Storybook does not detect the gradient color and treats it as transparent, which is why it reports this issue. I verified this in the first screenshot.


<img width="602" height="119" alt="Image" src="https://github.com/user-attachments/assets/2cafca84-3239-490c-b610-f9a3a12a9b3e" />


This issue is not reproducible in the Axe browser extension, as shown in the second screenshot.

<img width="720" height="312" alt="Image" src="https://github.com/user-attachments/assets/ee3bb752-b09a-4397-beab-ebb033d6cf33" />

---


- **Issue 2** -- https://github.com/cengage/react-magma/issues/1856#issuecomment-3371638183 ✅
    There are no accessibility issues reported for Pagination in the Accessibility tab for either v4 or v5.

<img width="1774" height="1170" alt="Image" src="https://github.com/user-attachments/assets/eab7b017-93dc-45cb-8b31-f10ca0d11a79" />

---


- **Issue 3** -- https://github.com/cengage/react-magma/issues/1856#issuecomment-3371645685 ✅
     There are no accessibility issues reported for TimePicker in the Accessibility tab for either v4 or v5.

<img width="714" height="606" alt="Image" src="https://github.com/user-attachments/assets/f884f319-d516-419c-8fce-73925c0b44d4" />

---


- **Issue 4** -- https://github.com/cengage/react-magma/issues/1856#issuecomment-3371707150 ✅
     There are no accessibility issues reported for Select in the Accessibility tab for either v4 or v5.

<img width="760" height="567" alt="Image" src="https://github.com/user-attachments/assets/2c02f345-1f2b-41cf-aa85-bc708825471a" />

---


- **Issue 5** -- https://github.com/cengage/react-magma/issues/1856#issuecomment-3371710982 ✅
     There are no accessibility issues reported for Select in the Accessibility tab for either v4 or v5.

<img width="764" height="556" alt="Image" src="https://github.com/user-attachments/assets/5bbc86e8-a183-4d23-be1a-5e9281904a6b" />

---


- **Issue 6** -- https://github.com/cengage/react-magma/issues/1856#issuecomment-3371762609 ✅
    There are no accessibility issues reported for NativeSelect in the Accessibility tab for either v4 or v5.


<img width="686" height="573" alt="Image" src="https://github.com/user-attachments/assets/248ac90d-0aa2-45a4-9ca9-f532e51efd9d" />

---


- **Issue 7** -- https://github.com/cengage/react-magma/issues/1856#issuecomment-3372357497 ✅
    The text and background contrast is the same as in previous elements. Also, when checking the accessibility values in DevTools, everything looks correct. Additionally, the regular Axe extension does not reproduce this issue. 

<img width="351" height="367" alt="Image" src="https://github.com/user-attachments/assets/d1b97437-252b-41a4-9557-02ac3fa40455" />

---


- **Issue 8** -- https://github.com/cengage/react-magma/issues/1856#issuecomment-3372383712 ❌ 🚀
   Found a valid issue; the rest aren't really related to the Input and aren't caught by standard axe.

   `role="img" alt text - role-img-alt`
Any element with `role="img"` requires alt text.
Proposed fix: add `aria-hidden="true"` if it's a decorative icon inside the input.

---


- **Issue 9** -- https://github.com/cengage/react-magma/issues/1856#issuecomment-3372387595 ❌ 🚀
   Time Picker - missing label
Element does not have an implicit (wrapped) `label`.
Proposed fix: just update the example - replace `p` with `label`.

---


- **Issue 10** -- https://github.com/cengage/react-magma/issues/1856#issuecomment-3375674655 ❓ ✅
      The issue is related to the dropdown, not the table. Plus I'm not sure we want to fix it since it's not caught by axe - only flagged inside Storybook itself.

---
**Part 2**
---


- **Issue 1** -- https://github.com/cengage/react-magma/issues/1857#issuecomment-3376706113 ❓✅
     Only on v4
     I'm not sure we want to fix it since it's not caught by axe - only flagged inside Storybook itself.


---

- **Issue 2** -- https://github.com/cengage/react-magma/issues/1857#issuecomment-3376715186 ❓ ✅
     I'm not sure we want to fix it since it's not caught by axe - only flagged inside Storybook itself.

---

- **Issue 3** -- https://github.com/cengage/react-magma/issues/1857#issuecomment-3376725108❓ ✅
    The issue is related to the `TextArea`. Plus I'm not sure we want to fix it since it's not caught by axe - only flagged inside Storybook itself.

--- 

- **Issue 4** -- https://github.com/cengage/react-magma/issues/1857#issuecomment-3376736898 ❌ 🚀
    The reported issue could not be reproduced, however I found a separate accessibility issue in the Selectable And Sortable example (see screenshot for details). The problem is likely in the table component itself rather than the example. Worth fixing.

<img width="533" height="373" alt="Image" src="https://github.com/user-attachments/assets/631cbf2e-d109-418e-9ee5-66a5e9fbf078" />

---

- **Issue 5** -- https://github.com/cengage/react-magma/issues/1857#issuecomment-3376810072  ✅ (⏰ / 🚀)
     The issue reproduces, but it's likely a Storybook's axe cannot see the list elements properly and flags it as an error. Standard axe does not catch this issue. 

  - There is another issue: aria-valid-attr-value 

    In ScrollSpy examples, tabs were generating `aria-controls` pointing to t`abpanel-{id}-{index}` - an element that doesn't exist in the DOM. Instead of standard `TabPanel` components, ScrollSpy uses `<section>` tags with different ids (card1, card2, etc.). Screen reader couldn't find the associated panel → axe-core was throwing `aria-valid-attr-value` violation.

---
Part 4
---


- **Issue 1** -- https://github.com/cengage/react-magma/issues/1859#issuecomment-3385834945 ✅ 
   There are no accessibility issues reported for PasswordInput in the Accessibility tab for either v4 or v5.


<img width="713" height="575" alt="Image" src="https://github.com/user-attachments/assets/a7269de4-57da-424a-ae89-492e75bab6f3" />


---

- **Issue 2** -- https://github.com/cengage/react-magma/issues/1859#issuecomment-3385837776 ✅ 
   There are no accessibility issues reported for PasswordInput in the Accessibility tab for either v4 or v5.

---

- **Issue 3** -- https://github.com/cengage/react-magma/issues/1859#issuecomment-3385843123 ❌ 🚀
   The issue is reproducible and needs to be fixed.


---


- **Issue 4** -- https://github.com/cengage/react-magma/issues/1859#issuecomment-3385880486 ❓
    This is a strange issue. It is not detected by the regular Axe extension, and I do not see any obvious contrast problems. However, it may be worth taking a closer look to understand why this issue is being reported.


---
    

- **Issue 5** -- https://github.com/cengage/react-magma/issues/1859#issuecomment-3385899926 ❌ 🚀
      The issue is reproducible and needs to be fixed.


---


- **Issue 6** -- https://github.com/cengage/react-magma/issues/1859#issuecomment-3385935835 ❌ ⏰
     The issue is reproducible and needs to be fixed.

**Comment by serhiiyanhol-cengage** [comment]
@artemtrusov-cengage 

v4/v5: [all inputs / custom topics row story](https://github.com/cengage/react-magma/issues/1856#issuecomment-3372383712)
**macos**: 
<img width="1616" height="421" alt="Image" src="https://github.com/user-attachments/assets/58a00cab-e5c8-410f-a167-e5a6d60ae61f" />
**windows**:
<img width="2011" height="755" alt="Image" src="https://github.com/user-attachments/assets/13cf8a8f-4fb2-4eac-8d05-b211b599656f" />

v4/v5 both OS, all browsers [tabs](https://github.com/cengage/react-magma/issues/1857#issuecomment-3376810072)
<img width="1750" height="526" alt="Image" src="https://github.com/user-attachments/assets/2b5903e5-813d-472a-bbfb-2c0798f77a33" />

v4/v5 both OS, all browsers [ToggleButtonGroup](https://github.com/cengage/react-magma/issues/1859#issuecomment-3385899926)
reported accessibility violation was fixed but another one appeared
<img width="1609" height="485" alt="Image" src="https://github.com/user-attachments/assets/657e6108-ee88-4585-97a4-6a98e833f98e" />

v4/v5 both OS, all browsers [treeview](https://github.com/cengage/react-magma/issues/1859#issuecomment-3385935835)
<img width="1913" height="224" alt="Image" src="https://github.com/user-attachments/assets/d0273799-ed2e-48e4-90ab-6a5b4d805088" />
<img width="1912" height="143" alt="Image" src="https://github.com/user-attachments/assets/dd4605d4-f7e9-49f5-89c1-349120b3beda" />
